sdk/clientcore/core/src/main/java/io/clientcore/core/serialization/json/JsonSerializer.java

```diff
@@ -89,6 +86,41 @@ public <T> T deserializeFromBytes(byte[] bytes, Type type) throws IOException {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private <T> T deserializeListOfBinaryData(JsonReader jsonReader) throws IOException {
+        JsonToken token = jsonReader.currentToken();
+        if (token == null) {
+            token = jsonReader.nextToken();
+        }
+
+        if (token != JsonToken.START_ARRAY) {
+            throw LOGGER.throwableAtError()
+                .addKeyValue("token", token)
+                .log("Unexpected token to begin an array", IllegalStateException::new);
+        }
+
+        List<BinaryData> list = new ArrayList<>();
+        while (jsonReader.nextToken() != JsonToken.END_ARRAY) {
+            list.add(BinaryData.fromObject(jsonReader.readUntyped()));
+        }
+        return (T) list;
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> T deserializeListOfJsonSerializables(JsonReader jsonReader, ParameterizedType parameterizedType)
+        throws IOException {
+        List<?> list = jsonReader.readArray(arrayReader -> {
+            Type actualTypeArgument = parameterizedType.getActualTypeArguments()[0];
+            Class<?> clazz = (Class<?>) actualTypeArgument;
+            try {
+                return clazz.getMethod("fromJson", JsonReader.class).invoke(null, arrayReader);
```

Comment:
Why are we looking this up from within the read loop? The type can't change here.

Also, use a `MethodHandle` here as it should incur less overhead.